### ISSUE TITLE
Unable to upgrade to new jx version

### ISSUE THREAD

**Issue by gmacario** [jenkins-x/jx#1993]
### Summary 

While executing `jx version` I was asked whether I would like to upgrade to the new jx version.
If I answer "Yes", the programs downloads the new version but fails when renaming `/tmp/jx` to `~/bin/jx` where the jx binary was originally installed.

### Steps to reproduce the behavior

Type `jx version`, the `Yes` when asked:

```
gmacario@cloudshell:~ (kubernetes-workshop-218213)$ jx version
Using helmBinary helm with feature flag: none
NAME               VERSION
jx                 1.3.431
jenkins x platform 0.0.2765
Kubernetes cluster v1.9.7-gke.6
kubectl            v1.10.7
helm client        v2.10.0+g9ad53aa
helm server        v2.10.0+g9ad53aa
git                git version 2.11.0

A new jx version is available: 1.3.436
? Would you like to upgrade to the new jx version? Yes
Downloading https://github.com/jenkins-x/jx/releases/download/v1.3.436/jx-linux-amd64.tar.gz to /home/gmacario/.jx/bin/jx.tgz...
Downloaded /home/gmacario/.jx/bin/jx.tgz
error: rename /tmp/jx /home/gmacario/.jx/bin/jx: invalid cross-device link
gmacario@cloudshell:~ (kubernetes-workshop-218213)$
```

### Jx version

The output of `jx version` is:

```
Using helmBinary helm with feature flag: none
NAME               VERSION
jx                 1.3.431
jenkins x platform 0.0.2765
Kubernetes cluster v1.9.7-gke.6
kubectl            v1.10.7
helm client        v2.10.0+g9ad53aa
helm server        v2.10.0+g9ad53aa
git                git version 2.11.0
```

### Kubernetes cluster

What kind of Kubernetes cluster are you using & how did you create it?

Google Cloud Platform

### Operating system / Environment

Google Cloud Shell

### Actual behavior

```
A new jx version is available: 1.3.436
? Would you like to upgrade to the new jx version? Yes
Downloading https://github.com/jenkins-x/jx/releases/download/v1.3.436/jx-linux-amd64.tar.gz to /home/gmacario/.jx/bin/jx.tgz...
Downloaded /home/gmacario/.jx/bin/jx.tgz
error: rename /tmp/jx /home/gmacario/.jx/bin/jx: invalid cross-device link
gmacario@cloudshell:~ (kubernetes-workshop-218213)$
```

**Comment by gmacario** [comment]
The error message 

```
error: rename /tmp/jx /home/gmacario/.jx/bin/jx: invalid cross-device link
gmacario@cloudshell:~ (kubernetes-workshop-218213)$
```

is self-explanatory

```
gmacario@cloudshell:~ (kubernetes-workshop-218213)$ df -h /tmp /home/gmacario/.jx
Filesystem      Size  Used Avail Use% Mounted on
overlay          31G   25G  5.9G  81% /
/dev/sdb1       4.8G  299M  4.3G   7% /home
gmacario@cloudshell:~ (kubernetes-workshop-218213)$
```

**Comment by gmacario** [comment]
The bug seems quite trivial to fix (basically doing a "cp" and "rm" instead then a "mv"), but I am little naive with golang and jx architecture.

If anybody can provide me with some guidance I can give it a try and submit a PR as part of my [Hacktoberfest 2018](https://hacktoberfest.digitalocean.com/) contribution...

**Comment by gmacario** [comment]
As a matter of fact, if I am not mistaken the fix is already available in https://github.com/jenkins-x/jx/pull/1950

**Comment by gmacario** [comment]
Closing - this is actually a duplicate of https://github.com/jenkins-x/jx/issues/1893#issuecomment-429012403

**Comment by gmacario** [comment]
Testing jx upgrade now that #1950 has been merged to master

```
gmacario@cloudshell:~ (kubernetes-workshop-218213)$ jx version
Using helmBinary helm with feature flag: none
NAME               VERSION
jx                 1.3.457
jenkins x platform 0.0.2765
Kubernetes cluster v1.9.7-gke.6
kubectl            v1.10.7
helm client        v2.10.0+g9ad53aa
helm server        v2.10.0+g9ad53aa
git                git version 2.11.0

A new jx version is available: 1.3.458
? Would you like to upgrade to the new jx version? Yes

Downloading https://github.com/jenkins-x/jx/releases/download/v1.3.458/jx-linux-amd64.tar.gz to /home/gmacario/.jx/bin/jx.tgz...
Downloaded /home/gmacario/.jx/bin/jx.tgz
Jenkins X client has been installed into /home/gmacario/.jx/bin/jx
gmacario@cloudshell:~ (kubernetes-workshop-218213)$
```
